tests/std/tests/Dev10_500860_overloaded_address_of/test.cpp

```diff
@@ -42,6 +42,12 @@ namespace Meow {
 
         constexpr Evil() noexcept : data(1701) {}
 
+        template <int = 0>
+        const Evil& operator=(const Evil&) const {

```

Comment:
Is this operator= tested at all?

Reply:
This `operator=` is supposed to make the existing test work, not to test the new `operator=` and `swap` overloads (which are tested in `P2321R2_proxy_reference`).

This test explicitly instantiates a `tuple` of `Evil`s:

https://github.com/microsoft/STL/blob/6a4351b820d7728512ccf5a8ff064fbd018db721/tests/std/tests/Dev10_500860_overloaded_address_of/test.cpp#L116

This causes the instantiation of every (non-template) function in `tuple`, including the const-qualified `swap`. To make this `swap` work, it must be possible to swap `const Evil` lvalues.

I decided to add a const-qualified `operator=` to enable `std::swap` instead of adding a non-member `swap(const Evil&, const Evil&)`, because this test also exercises `std::swap` and I don't want to add another `swap` to the overload set (though it shouldn't actually matter).